### DISCUSSION TITLE
Global .tool-versions overwrite local .tool-versions when using source_up

### DISCUSSION THREAD

**Discussion by marcelmindemann** [asdf-community/asdf-direnv#91] [Q&A] [answered]
Hi,
I am using 
* asdf v0.9.0
* direnv 2.30.3
* zsh 5.8

on macOS.

Let's say I want to globally use Python 3.10, but in a specific project at `~/myproject` I want to use the older Python 3.8. I also have some important environment variables I want to use globally, for example `PYPI_APIKEY`.

If I want to use asdf-direnv to automatically update my $PATH depending on whether I am in `~` or `~/myproject`, I need the following files in my `~`:

`~/.tool-versions`
```
python 3.10.2
direnv 2.30.3
```
`~/.envrc`
```
use asdf
export PYPI_APIKEY=xxx
```

In my project at `~/myproject` I need:
`~/myproject/.tool-versions`:
```
python 3.9.10
```
`~/myproject/.envrc`:
```
use asdf
source_up
```

You can see where this leads to: Since `source_up` loads `.envrc` files going up the directory tree, python 3.10 overwrites my local python 3.9. So the global python version overwrites the local python version, which is counter-intuitive. Rather, the `.envrc` files should be resolved _downwards_ the directory tree so that deeper files overwrite their parents.

Is there a workaround we could use to fix this behaviour? The only thing I can think of is not having `use asdf` in our `~/.envrc`, but then we miss out on most of the benefits of asdf-direnv alltogether.

**Comment by ghost** [comment, marked as answer]
If you change the order of the commands, you get the expected result

```
source_up
use asdf
```
